File: amazon-scraper/python/relatedSearches.py
import urllib.parse
import requests
from bs4 import BeautifulSoup

# Configuration
token = "<SDO-token>"
search_query = "laptop stands"  # Change this to any search term
geocode = "us"
zipcode = "10001"

# Build search URL
target_url = f"https://www.amazon.com/s?k={urllib.parse.quote_plus(search_query)}"

targetUrl = urllib.parse.quote(target_url)
apiUrl = "https://api.scrape.do/plugin/amazon/?token={}&url={}&geocode={}&zipcode={}&output=html".format(token, targetUrl, geocode, zipcode)
response = requests.request("GET", apiUrl)

soup = BeautifulSoup(response.text, "html.parser")

# Find all related search terms
related_searches = []

for div in soup.find_all("div", class_="a-box-inner a-padding-mini"):
    text = div.get_text(strip=True)
    if text:
        related_searches.append(text)

print(f"Related searches:")
for i, term in enumerate(related_searches, 1):
    print(f"  {i}. {term}")

File: amazon-scraper/node.js/relatedSearches.js
const axios = require('axios');
const cheerio = require('cheerio');

// Configuration
const token = "<SDO-token>";
const searchQuery = "laptop stands"; // Change this to any search term
const geocode = "us";
const zipcode = "10001";

// Build search URL
const targetUrl = `https://www.amazon.com/s?k=${encodeURIComponent(searchQuery)}`;

// Make API request
const encodedUrl = encodeURIComponent(targetUrl);
const apiUrl = `https://api.scrape.do/plugin/amazon/?token=${token}&url=${encodedUrl}&geocode=${geocode}&zipcode=${zipcode}&output=html`;

async function scrapeRelatedSearches() {
    const response = await axios.get(apiUrl);
    const $ = cheerio.load(response.data);

    // Find all related search terms
    const relatedSearches = [];

    $('div.a-box-inner.a-padding-mini').each((index, element) => {
        const text = $(element).text().trim();
        if (text) {
            relatedSearches.push(text);
        }
    });

    console.log('Related searches:');
    relatedSearches.forEach((term, index) => {
        console.log(`  ${index + 1}. ${term}`);
    });
}

scrapeRelatedSearches().catch(console.error);
